pkg/deepcopy/deepcopy_integration_test.go

```diff
@@ -86,9 +86,30 @@ var _ = Describe("CRD Generation From Parsing to CustomResourceDefinition", func
 		Expect(err).NotTo(HaveOccurred())
 		rt.OutputRules = genall.OutputRules{Default: output}
 
+		// TODO(akutz) This did not used to be necessary for some reason, as it
+		//             seems pkg/loader used to auto-magically act against the
+		//             working directory, even if not roots were specified. this
+		//             behavior is not documented anywhere, and I was torn as to
+		//             put the commented-out code *here* or update the loader to
+		//             assume roots=["."] if len(roots) == 0. For the sake of
+		//             backwards-compatibility I opted for the latter.
```

Comment:
I think defaulting to `.` when unset makes sense (and changing that might break ppl).

Edit: Now I read the other comment, so this never worked when using the `cli`?

Reply:
It was not documented as such, no. The flag is not omitted from any example:

```bash
	# Generate RBAC manifests and crds for all types under apis/,
	# outputting crds to /tmp/crds and everything else to stdout
	controller-gen rbac:roleName=<role name> crd paths=./apis/... output:crd:dir=/tmp/crds output:stdout

	# Generate deepcopy/runtime.Object implementations for a particular file
	controller-gen object paths=./apis/v1beta1/some_types.go

	# Generate OpenAPI v3 schemas for API packages and merge them into existing CRD manifests
	controller-gen schemapatch:manifests=./manifests output:dir=./manifests paths=./pkg/apis/... 

	# Run all the generators for a given project
	controller-gen paths=./apis/...
```

or shown as optional in help:

```bash
+paths package
	=<[]string>  represents paths and go-style path patterns to use as package roots.
```

I've traced out how the paths are ingested, and they do allow them to be empty since they're not a proper flag. They come this way:

1. the command line options are parsed from https://github.com/kubernetes-sigs/controller-tools/blob/cb13ac551a0599044e50ed756735a8f438a24631/cmd/controller-gen/main.go#L168
2. which creates a set of proto, runtime options at https://github.com/kubernetes-sigs/controller-tools/blob/cb13ac551a0599044e50ed756735a8f438a24631/pkg/genall/options.go#L76
3. which parses the command line `paths` optoin at https://github.com/kubernetes-sigs/controller-tools/blob/cb13ac551a0599044e50ed756735a8f438a24631/pkg/genall/options.go#L149
4. which get returned and read into the `ForRoots` function at https://github.com/kubernetes-sigs/controller-tools/blob/cb13ac551a0599044e50ed756735a8f438a24631/pkg/genall/options.go#L82
5. which is what calls `loader.LoadRoots` at https://github.com/kubernetes-sigs/controller-tools/blob/cb13ac551a0599044e50ed756735a8f438a24631/pkg/genall/genall.go#L211